### PR TITLE
Add URL constructor

### PR DESCRIPTION
**Pull request by turbolent**
See https://url.spec.whatwg.org/#api and 
https://developer.mozilla.org/en-US/docs/Web/API/URL/URL

### CODE REVIEW

**Comment by sjrd** [comment]
Given that the `URL` constructor does not seem to be supported in IE yet, this should go [in the `experimental` package](https://github.com/scala-js/scala-js-dom/tree/master/src/main/scala/org/scalajs/dom/experimental), in a file `URL.scala`. It's weird that the `object URL` and `class URL` won't live in the same package, but that reflects their respective level of support.

**Comment by sjrd** [comment]
That's all.

**Comment by turbolent** [comment]
Thanks for the feedback! Moved the class to the experimental package as suggested.

File: src/main/scala/org/scalajs/dom/experimental/URL.scala [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    */
+  var search: String = js.native
+
+  /**
+    * Is a DOMString containing a '#' followed by the fragment identifier of the URL.
+    *
+    * MDN
+    */
+  var hash: String = js.native
+}
```
Comment (sjrd) [inline src/main/scala/org/scalajs/dom/experimental/URL.scala]: There's a missing EOL at the end of the file here.

**Comment by sjrd** [comment]
Otherwise good for me. Could you squash your commits into a single commit, please?

**Comment by turbolent** [comment]
Squashed. You can also now do this straight from GitHub BTW: https://help.github.com/articles/about-pull-request-merge-squashing/

**Comment by sjrd** [comment]
LGTM. 

GitHub does not let me squash your commits *and* create a merge commit for merging the PR, unfortunately.

**Pull request merged**